off-chain/test/Examples/Addition/Actions.purs

```diff
@@ -82,6 +89,20 @@ handleAction userAction lockedValue getScriptUtxo =
           "Unexpected set of UTXOs at script addres. Should be only one UTXO, but got:\n"
             <> show scriptUtxos
 
+queryBlockchainState :: Contract (UtxoMap /\ AdditionState)
+queryBlockchainState = do
+  scriptUtxos <- getScriptUtxosFromChain
+  (_ /\ output) <-
+    liftMaybe
+      (error "can't find a single utxo in the script")
+      $ head (toUnfoldable scriptUtxos)
+  (AdditionDatum record) <-
+    case view _datum $ view _output output of
```

Comment:
Maybe `getTypedDatum` from `Seath.Test.Examples.Utils` can be extended to `forall a. FromData a => TransactionOutputWithRefScript -> Either String a` instead of `forall a. FromData a => TransactionOutputWithRefScript -> Maybe a` to get error for both cases (no datum and decode error) and used here with `liftEither`